src/net.cpp

```diff
@@ -90,6 +90,9 @@ std::map<CNetAddr, LocalServiceInfo> mapLocalHost GUARDED_BY(cs_mapLocalHost);
 static bool vfLimited[NET_MAX] GUARDED_BY(cs_mapLocalHost) = {};
 std::string strSubVersion;
 
+extern RecursiveMutex cs_main;
```

Comment:
It seems like kind of a layer violation to import these here, but, I don't know a better solution either.

Reply:
Agree that it is a layer violation. If I included it via the header, it would be a circular dependency according to some linter we run. But the same pattern is used in a lot of other places, see `git grep 'extern RecursiveMutex cs_main;'`. Those philosophical locking "issues" and code dependencies/layer violations can be fixed later, if needed.

Reply:
This is pretty ugly, but I see it may be needed. Agreed with fixing it this way for now.

FWIW, IMO having an 'extern' like this is as much a circular dependency as what you're trying to avoid, just one that isn't detected by the linter.

Reply:
At least with `extern` we can avoid including the full header and littering the namespace here ;)

Reply:
Fixed